rcl/src/rcl/publisher.c

```diff
@@ -22,23 +22,17 @@ extern "C"
 #include <stdio.h>
 #include <string.h>
 
-#include "./common.h"
 #include "rcl/allocator.h"
 #include "rcl/error_handling.h"
 #include "rcl/expand_topic_name.h"
 #include "rcl/remap.h"
 #include "rcutils/logging_macros.h"
 #include "rmw/error_handling.h"
-#include "rmw/rmw.h"
 #include "rmw/validate_full_topic_name.h"
 
-typedef struct rcl_publisher_impl_t
-{
-  rcl_publisher_options_t options;
-  rmw_qos_profile_t actual_qos;
-  rcl_context_t * context;
-  rmw_publisher_t * rmw_handle;
-} rcl_publisher_impl_t;
+#include "./common.h"
+#include "./publisher_impl.h"
+
```

Comment:
sneaky extra line so I'll mention it...

nitpick: minimize vertical whitespace

There's another like it below.

Reply:
Removed